source/adapters/cuda/kernel.cpp

```diff
@@ -377,3 +379,41 @@ urKernelSetArgSampler(ur_kernel_handle_t hKernel, uint32_t argIndex,
   }
   return Result;
 }
+
+UR_APIEXPORT ur_result_t UR_APICALL urKernelGetSuggestedLocalWorkSize(
+    ur_kernel_handle_t hKernel, ur_queue_handle_t hQueue, uint32_t workDim,
+    const size_t *pGlobalWorkOffset, const size_t *pGlobalWorkSize,
+    size_t *pSuggestedLocalWorkSize) {
+  // Preconditions
+  UR_ASSERT(hQueue->getContext() == hKernel->getContext(),
+            UR_RESULT_ERROR_INVALID_KERNEL);
+  UR_ASSERT(workDim > 0, UR_RESULT_ERROR_INVALID_WORK_DIMENSION);
+  UR_ASSERT(workDim < 4, UR_RESULT_ERROR_INVALID_WORK_DIMENSION);
+
+  std::ignore = pGlobalWorkOffset;
+  ur_context_handle_t Context = hQueue->getContext();
+  ur_device_handle_t Device = hQueue->Device;
+  ur_result_t Result = UR_RESULT_SUCCESS;
+  size_t MaxWorkGroupSize = 0u;
+  size_t ThreadsPerBlock[3] = {};
```

Comment:
We don't need this variable. We should just pass `pSuggestedLocalWorkSize` into `guessLocalWorkSize`

Reply:
I respectfully disagree with you here. As I see from https://github.com/oneapi-src/unified-runtime/blob/befdc7cdf8e5c333092ba08684372421d52faaa7/source/ur/ur.hpp#L352, the function requires a 3d array, and we cannot ensure the `pSuggestedLocalWorkSize` points to a 3d array.